modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuildController.java

```diff
@@ -184,60 +233,113 @@ private CompletableFuture<?> onIndexRemoved(RemoveIndexEventParameters parameter
 
     private CompletableFuture<?> onPrimaryReplicaElected(PrimaryReplicaEventParameters parameters) {
         return inBusyLockAsync(busyLock, () -> {
-            TablePartitionId primaryReplicaId = (TablePartitionId) parameters.groupId();
-
             if (isLocalNode(clusterService, parameters.leaseholderId())) {
-                primaryReplicaIds.add(primaryReplicaId);
+                // TODO https://issues.apache.org/jira/browse/IGNITE-24375
+                // Need to remove TablePartitionId check here and below.
+                assert parameters.groupId() instanceof ZonePartitionId || parameters.groupId() instanceof TablePartitionId :
+                        "Primary replica ID must be of type ZonePartitionId or TablePartitionId";
```

Comment:
please add parameters.grouprId() to error message

Reply:
Thanks. Fixed.